accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,242 @@
+
+## Teach reposync on how to use Cloud RMT authentication mechanism
+
+Reposync is already able to deal with the basic authentication mechanism. It receives the id of a `suseCredentials` record and loads the basic authentication data.
+This mechanism will be enhanced to consider also the new column with the authentication headers. The following changes need to be implemented:
+  - Modify method [`_url_with_repo_credentials`](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py#L1785) to also load and return the http authentication headers from table `suseCredentials`
+  - Enhance [repository plugins](https://github.com/uyuni-project/uyuni/tree/master/backend/satellite_tools/repo_plugins) to receive http headers
+  - Enhance reposync [zypper plugin](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/spacewalk-extra-http-headers) to receive http header. Header will be written to a temporary file in the same location of zypper `.repo` file.
+
+Another mechanism for [authentication headers](https://github.com/uyuni-project/uyuni/pull/2136) is already in place which loads headers from a configuration file. We should cope with this implementation and merger header loaded form both locations.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Uyuni/SUSE manager pay-as-you-go
+
+We could define a uyuni/suse manager pay-as-you-go image with access to all cloud RMT repositories.
+It would be possible to synchronize any product directly from cloud in a more simple and straightforward way.
+
+**Drawbacks:**
+- User will have access to repository and product he is not paying for.
+- Will not be suitable for scenarios were Uyuni is on-perm but managing cloud instances
+
+## Reposync only metadata
+
+Change reposync and SCC to allow metadata access without SCC account to load repositories metadata.
+Pay-as-you-go client could download data from cloud RMT server directly.
+This will be a huge change, not sure if it is even possible to all providers.
+Possible security impact of having repository metadata publicly accessible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- What happens and how to deal if the pay-as-you-go instance we loaded the data from get terminated or stopped?
+  - Possible approach is remove the affected cloudrmt auth configs from the Server. Channels will be kept with all the existing content and assigned server will still see it. But no new updates will added to the channels.
+
+- Should we have a UI showing all the pay-as-you-go authentications registered and allow user to remove existing ones?
+
+- In case of multiple options, what should be preferred? I think it also depends on where Uyuni Server is running. If it is running in the cloud, the local RMT servers should be preferred, while if the Server is running outside of the Cloud, SCC should be preferred.
```

Comment:
I'd suggest to always prefer the SCC CDN if available out of reliability concerns.

Reply:
I agree with you. I remove it from unresolved questions and added it implementation requirement when adding a product.